Process_Library/GOCART2G_MieMod.F90

```diff
@@ -326,73 +330,88 @@ type(GOCART2G_Mie) function GOCART2G_MieCreate ( rcfile, wavelengths, nmom, rc )
+      allocate (this%vol(this%nbin),__NF_STAT__)
+      allocate (this%area(this%nbin),__NF_STAT__)
+
+      do j = 1, this%nbin
+        this%reff(j)%rh = reff_table(:,j)
+        !Insert growth factor
+        this%gf(j)%rh   = gf_table(:,j)
+        !Wet particle density [kg m-3]
+        this%rhop(j)%rh = rhop_table(:,j)
+        !Dry particle density [kg m-3]
+        this%rhod(j)%rh = rhod_table(:,j)
+        !Volume [m3 kg-1]
+        this%vol(j)%rh  = vol_table(:,j)
+        !Area [m2 kg-1]
+        this%area(j)%rh = area_table(:,j)
+      enddo
+
+      allocate (this%bext(this%nch,this%nbin), __NF_STAT__)
+      allocate (this%bsca(this%nch,this%nbin), __NF_STAT__)
+      allocate (this%bbck(this%nch,this%nbin), __NF_STAT__)
+      allocate (this%g(this%nch,this%nbin),    __NF_STAT__)
+      allocate (this%pback(this%nch,this%nbin,this%nPol),    __NF_STAT__)
+      if ( nmom_ > 0 ) then
+         allocate (this%pmom(this%nch,this%nbin,this%nMom,this%nPol),    __NF_STAT__)
+      end if
+      allocate (this%refr(this%nch,this%nbin), __NF_STAT__)
+      allocate (this%refi(this%nch,this%nbin), __NF_STAT__)
 
 !     Now we linearly interpolate the input table to the output table grid
 !     of requested channels
       do j = 1, this%nbin
-       do i = 1, this%nrh
         do n = 1, this%nch
-         call polint(channels_table,bext_table(:,i,j),nch_table, &
-                     this%wavelengths(n),this%bext(i,n,j),yerr)
-         call polint(channels_table,bsca_table(:,i,j),nch_table, &
-                     this%wavelengths(n),this%bsca(i,n,j),yerr)
-         call polint(channels_table,bbck_table(:,i,j),nch_table, &
-                     this%wavelengths(n),this%bbck(i,n,j),yerr)
-         call polint(channels_table,g_table(:,i,j),nch_table,    &
-                     this%wavelengths(n),this%g(i,n,j),yerr)
-         call polint(channels_table,refr_table(:,i,j),nch_table, &
-                     this%wavelengths(n),this%refr(i,n,j),yerr)
-         call polint(channels_table,refi_table(:,i,j),nch_table, &
-                     this%wavelengths(n),this%refi(i,n,j),yerr)
-         do ipol = 1, this%nPol
-                  call polint(channels_table,pback_table(:,i,j,ipol),nch_table,    &
-                       this%wavelengths(n),this%pback(i,n,j,ipol),yerr)
-         end do
-         if ( nmom_ > 0 ) then
+          allocate (this%bext(n,j)%rh(this%nrh), __NF_STAT__)
+          allocate (this%bsca(n,j)%rh(this%nrh), __NF_STAT__)
+          allocate (this%bbck(n,j)%rh(this%nrh), __NF_STAT__)
+          allocate (this%g(n,j)%rh(this%nrh), __NF_STAT__)
+          allocate (this%refr(n,j)%rh(this%nrh), __NF_STAT__)
+          allocate (this%refi(n,j)%rh(this%nrh), __NF_STAT__)
+          do i = 1, this%nrh
+            call polint(channels_table,bext_table(:,i,j),nch_table, &
+                     this%wavelengths(n),this%bext(n,j)%rh(i),yerr)
+            call polint(channels_table,bsca_table(:,i,j),nch_table, &
```

Comment:
How ugly would it be to move the `i` index to inside the polint procedure?

Reply:
it is easy. Done